### PR TITLE
Elastic search

### PR DESCRIPTION
**Pull request by gislenius**

### CODE REVIEW

**Comment by gislenius** [comment]
All fields: http://federico.nttdata.28.io/v1/_queries/public/entities-2.jq

**Comment by gislenius** [comment]
Mapping: http://xbrl-japan.28.io/es-mapping.jq

File: tasks/28.js [hunk 1]
```diff
@@ -276,39 +282,44 @@ var runQueries = function(projectName, sequenceOfQueriesToRun) {
...
-    Config.$28.createDatasource(projectName, datasource.category, datasource.name, projectToken, difault, JSON.stringify(datasource.credentials))
-        .then(function(){
-            $.util.log(datasource.name + ' created');
-            defered.resolve(credentials);
-        })
-        .catch(function (error) {
-            $.util.log('datasource creation failed: ' + datasource.name);
-            defered.reject(error);
-        });
+    setTimeout(function() { // @todo: temp fix until datasouces api allows concurrent updates
```
Comment (wcandillon) [inline tasks/28.js]: @dknochen Keep this hack as is for now. The other hack I suggested doesn't seem to work :)

File: queries/modules/io/28/modules/xbrl/profiles/japan/core.jq [hunk 2]
```diff
@@ -43,7 +47,32 @@ declare function japan:entities(
 
          mw:find($entities:col, { "Profiles.FSA.SecuritiesCode" : { "$in" : [ $securitiescode ] } }),
 
-         entities:entities($eid)
+         let $conn := es:connection("xbrl")
+         where exists($full-text)
+         for $result in es:search(
+           $conn,
+           "japanxbrl-8_8",
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/profiles/japan/core.jq]: should be $config:db-name

File: queries/modules/io/28/modules/xbrl/profiles/japan/core.jq [hunk 3]
```diff
@@ -43,7 +47,32 @@ declare function japan:entities(
...
+                  }
+                  default return {
+                    "SubmitterNameAlphabetic" : $full-text
+                  }
+              }
+            }
+         ).hits.hits[]._source
+         return
+           copy $r := $result
+           modify rename json $r.id as "_id"
```
Comment (dknochen) [inline queries/modules/io/28/modules/xbrl/profiles/japan/core.jq]: this is no longer needed:
image

**Comment by dknochen** [comment]
@gislenius this is ready for review except for documentation in the elastic search module.

I'm not so convinced, that the sync'ing should be done in the cellstore. Maybe, we should move this code to the infosetgenerator? What do you think?

Its nice to have the elastic search module almost complete in any case.

**Comment by gislenius** [comment]
@dknochen Thanks! Something is broken with the tests though.

**Comment by gislenius** [comment]
@wcandillon we're now all okay, and can merge when you have reviewed as well.

**Comment by wcandillon** [comment]
Reviewed, amazing stuff!

**Pull request merged**